disease/query.py

```diff
@@ -372,54 +380,87 @@ def _record_order(self, record: Dict) -> (int, str):
+                                 MatchType.CONCEPT_ID)
+
+        non_merged_match = None
 
         # check concept ID match
         record = self.db.get_record_by_id(query_str, case_sensitive=False)
         if record:
-            response['match_type'] = MatchType.CONCEPT_ID
-            response = self._add_merged_record(response, record['merge_ref'])
-            return response
+            if 'merge_ref' in record:
+                record = self.db.get_record_by_id(record['merge_ref'],
+                                                  case_sensitive=False,
+                                                  merge=True)
+                if record is None:
+                    logger.error(f"Merge ref lookup failed for ref"
+                                 f"{record['merge_ref']} in record"
+                                 f"{record['concept_id']} from query"
+                                 f"{query_str}")
+                    response['match_type'] = MatchType.NO_MATCH
+                    return response
+                return self._add_vod(response, record, query,
+                                     MatchType.CONCEPT_ID)
+            else:
+                non_merged_match = (record, 'concept_id')
 
         # check other match types
         for match_type in ['label', 'alias']:
             # get matches list for match tier
             query_matches = self.db.get_records_by_type(query_str, match_type)
-            query_matches = [self.db.get_record_by_id(m['concept_id'], False)
+            query_matches = [self.db.get_record_by_id(m['concept_id'],
+                                                      case_sensitive=False)
                              for m in query_matches]
             query_matches.sort(key=self._record_order)
 
             # attempt merge ref resolution until successful
             for match in query_matches:
                 record = self.db.get_record_by_id(match['concept_id'], False)
                 if record:
-                    merge_ref = record['merge_ref']
-                    if merge_ref:
-                        response['match_type'] = MatchType[match_type.upper()]
-                        return self._add_merged_record(response, merge_ref)
+                    if 'merge_ref' in record:
+                        record = self.db.get_record_by_id(record['merge_ref'],
+                                                          case_sensitive=False,
+                                                          merge=True)
+                        if record is None:
+                            logger.error(f"Merge ref lookup failed for ref"
+                                         f"{record['merge_ref']} in record"
+                                         f"{record['concept_id']} from query"
+                                         f"{query_str}")
+                            response['match_type'] = MatchType.NO_MATCH
+                            return response
```

Comment:
I believe this is the exact as the same code above. I would place this in a new method and fix the logger.error message since merge_ref and concept_id keys won't be in a record if it's None.